js/src/inputs.jsx

```diff
@@ -29,10 +29,22 @@ export const ColorPicker = InputAdapter(Fluent.ColorPicker, (value, setValue) =>
   onChange: (e, v) => setValue(v.str),
 }), { policy: debounce, delay: 250 });
 
-export const ComboBox = InputAdapter(Fluent.ComboBox, (value, setValue) => ({
+export const ComboBox = InputAdapter(Fluent.ComboBox, (value, setValue, props) => ({
   selectedKey: value && value.key,
   text: value && value.text,
-  onChange: (e, option, i, text) => setValue(option || (text ? { text } : null)),
+  onChange: (e, option, i, text) => {
+    if (props.multiSelect) {
+      const options = new Set(props.options.map((item) => item.key));
+      let newValue = (Array.isArray(value) ? value : [value])
+        .filter((key) => options.has(key)); // Some options might have been removed.
+      newValue = option.selected
+        ? [...newValue, option.key]
+        : newValue.filter((key) => key !== option.key);
+      setValue(newValue);
```

Comment:
The code in this if branch is very similar to the code in `Dropdown` below. Perhaps we can make it a function? (Arguments: `props.options`, `option`; return `newValue`).